space_packet_parser/definitions.py

```diff
@@ -36,11 +34,112 @@ class XtcePacketDefinition:
 
     def __init__(
             self,
+            *,
+            sequence_container_list: Optional[list[packets.SequenceContainer]] = None,
```

Comment:
A thought here would be to add a larger allowable input and then go out to the `from_*` methods based on isinstance checks. This would potentially be backwards compatible too I think.

```suggestion
            self,
            xtce_document: Optional[Union[str, Path, TextIO]],
            *,
            sequence_container_list: Optional[list[packets.SequenceContainer]] = None,
            ...
            if isinstance(xtce_document, list):
                return self.from_sequence_container(...)
            elif isinstance(xtce_document, (str, Path, TextIO)):
                return self.from_document(...)
```

Reply:
Yeah, we could do that but I think I want to make this breaking change. The larger problem is that I want XtcePacketDefinition to be Python-only (no internal XML required) and have the `from_document` method turn the incoming XML into python objects that can directly instantiate XtcePacketDefinition and then throw away the original XML. 

Unfortunately currently there's an awkwardness in that we keep an XML tree object internal to XtcePacketDefinition when in theory we really shouldn't need it. The `_initialize_xml_tree` method that's currently necessary in `__init__` should really be a `to_xtce_document` method that only gets called when we want to output XML.

Reply:
> from_document method turn the incoming XML into python objects that can directly instantiate XtcePacketDefinition and then throw away the original XML.

Seems reasonable and I agree, we really want to just parse the XML document and never use it / search through it again. I think that could still work with the above instance checks, but I agree it does cause some awkwardness.

I'm trying to think of other package APIs that create objects and their is a lot of different preferences it seems.
```
ET.fromstring("xmlstring")
etree.parse("path-to-file")
pd.read_csv("path-to-csv")
```

I would think you'd want the list of sequence_containers to be an optional argument so you don't have to always remember to pass in the explicit keyword name though.

Also throwing out some other potential names/ideas for consideration:  class method`from_xml()`, root function in the package (like pd.read_csv which returns a DataFrame): `space_packet_parser.read_xml()` / `load_xml()`